src/resolve/resolved-dns-query.c

```diff
@@ -480,6 +480,14 @@ int dns_query_new(
 
         assert(m);
 
+        /* Check for records that is refused and refuse query for the records if matched in configuration */
+        DNS_QUESTION_FOREACH(key, question_utf8) {
+                if (set_contains(m->refuse_record_types, INT_TO_PTR(key->type))) {
+                        log_debug("Got request for %s record that is refused.", dns_type_to_string(key->type));
+                        return -ENOSYS;
```

Comment:
```c
return log_debug_errno(SYNTHETIC_ERRNO(ENOSYS), ...);
```

But, I do not think ENOSYS is not a good error code here. Please use more recognizable error code, e.g. `ENOANO` or so.

Reply:
@yuwata used ENOANO now